File: backend/src/main/java/dev/coldhands/pair/stairs/backend/infrastructure/persistance/entity/CombinationEntity.java
package dev.coldhands.pair.stairs.backend.infrastructure.persistance.entity;

import jakarta.persistence.*;

import java.util.List;

@Entity
public class CombinationEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(
            name = "combination_pair_member",
            joinColumns = @JoinColumn(name = "combination_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "pair_stream_id", referencedColumnName = "id")
    )
    private List<PairStreamEntity> pairs;

    protected CombinationEntity() {
    }

    public CombinationEntity(List<PairStreamEntity> pairs) {
        this.pairs = pairs;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public List<PairStreamEntity> getPairs() {
        return pairs;
    }

    public void setPairs(List<PairStreamEntity> pairs) {
        this.pairs = pairs;
    }
}

File: backend/src/main/kotlin/dev/coldhands/pair/stairs/backend/infrastructure/persistance/entity/CombinationEntity.kt
package dev.coldhands.pair.stairs.backend.infrastructure.persistance.entity

import jakarta.persistence.*

@Entity
@Table(name = "combinations")
class CombinationEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "combination_pair_member",
        joinColumns = [JoinColumn(name = "combination_id", referencedColumnName = "id")],
        inverseJoinColumns = [JoinColumn(name = "pair_stream_id", referencedColumnName = "id")]
    )
    val pairs: MutableList<PairStreamEntity>
) {
    // todo remove me once everything is kotlin
    constructor(pairs: List<PairStreamEntity>) : this(id = null, pairs = pairs.toMutableList())
}